src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,50 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private T[] elements;
 
+    public MateStack() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void push(T value) {
+        if (size == elements.length) {
+            elements = resize();
+        }
+        elements[size] = value;
+        size++;
```

Comment:
you can simplify the code
```suggestion
        elements[size++] = value;
```

Reply:
++